libs/common/types.hpp

```diff
@@ -119,6 +120,17 @@ namespace iroha {
 
       return b;
     }
+
+    static boost::optional<blob_t<size_>> from_hexstring(const std::string &data) {
+      if (data.size() != size_) {
```

Comment:
monadic bind?

Reply:
could you be more specific?

Reply:
I mean [this](https://github.com/hyperledger/iroha/blob/15cf6e3d46c8e18d992922143aa8075d21df1f36/libs/common/types.hpp#L212)
It's the most readable way to write functions returning optional

Reply:
I know what is monadic bind. Write example how can I use it.

Reply:
Well, maybe bind decrease the readabilty
smth like this:
```cpp
return boost::make_optional<>()
    | [&data, =size_](auto b) {
      if (data.size() != size_) return boost::none;
      return b;
    } | [](auto b) {..}
```